src/classes/RD2_ScheduleService.cls

```diff
@@ -411,6 +413,54 @@ public without sharing class RD2_ScheduleService {
         return nextDate;
     }
 
+    /***
+    * @description Calculates the date of the next donation and returns the Opportunity matcher record
+    * that contains the next donation installment and might contain the matching open Opportunity.
+    * @param rd Recurring Donation record
+    * @param rdSchedules Schedules on the Recurring Donation record
+    * @return RD2_OpportunityMatcher.Record
+    */
+    public RD2_OpportunityMatcher.Record getNextDonationDateRecord(npe03__Recurring_Donation__c rd, List<RecurringDonationSchedule__c> rdSchedules) {
```

Comment:
Let's revert back these changes from the Next Donation Date PR and try to use the Opp matcher parent branch code.